dex/pair/src/liquidity_pool.rs

```diff
@@ -10,148 +15,163 @@ const MINIMUM_LIQUIDITY: u64 = 1_000;
 pub trait LiquidityPoolModule:
     amm::AmmModule + config::ConfigModule + token_send::TokenSendModule
 {
-    fn pool_add_liquidity(
-        &self,
-        first_token_amount: BigUint,
-        second_token_amount: BigUint,
-    ) -> SCResult<BigUint> {
-        let first_token = self.first_token_id().get();
-        let second_token = self.second_token_id().get();
-        let total_supply = self.get_total_lp_token_supply();
-        let mut first_token_reserve = self.pair_reserve(&first_token).get();
-        let mut second_token_reserve = self.pair_reserve(&second_token).get();
+    fn pool_add_liquidity(&self, context: &mut AddLiquidityContext<Self::Api>) {
+        let zero = &BigUint::zero();
         let mut liquidity: BigUint;
 
-        if total_supply == 0 {
-            liquidity = core::cmp::min(first_token_amount.clone(), second_token_amount.clone());
-            let minimum_liquidity = BigUint::from(MINIMUM_LIQUIDITY);
-            require!(
-                liquidity > minimum_liquidity,
-                "First tokens needs to be greater than minimum liquidity"
+        if context.get_lp_token_supply() == zero {
```

Comment:
can't we do a separate function - like addInitialLiquidity - maybe protected at only_owner. and we can simply delete this if / else.

Reply:
we could but... would be easy from the SC point of view, but i think it'll complicate as we go further on the sw stack, service and frontend

Reply:
I do not think so. The first liquidity is added always from command line - not from frontend. I think it is safer to have a separate endpoint. It is part of the setup.

Reply:
And when we want to integrate the initial price discovery mechanism - it is mandatory to have a whitelisted address and a secure endpoint.

Reply:
will do on a separate PR :+1: